app/src/test/java/com/x8bit/bitwarden/ui/vault/feature/item/VaultItemScreenTest.kt

```diff
@@ -867,6 +867,10 @@ class VaultItemScreenTest : BaseComposeTest() {
             .onNodeWithText("Restore")
             .performClick()
 
+        mutableStateFlow.update {
+            it.copy(dialog = VaultItemState.DialogState.RestoreItemDialog)
```

Comment:
I would expect this test to just verify that the correct event is sent to the VM. There should be a different test to validate that the dialog is appropriated displayed based on state

Reply:
ahh I didn't even see the VM parts of the other tests, ok so just to be clear we are saying we can basically split the test, one to confirm the UI shows correctly based on the steps and one to verify the action is sent when the the lambda is fired.

Reply:
Yea, we don't want the screen test to know what the VM is doing. This currently implies that the VM will update the state.

So one test for verify the correct event is sent to the VM and a new separate test for testing the reaction to the sate change.

Reply:
done, updated the other tests/ added ones for separate checks